polkadot/runtime/parachains/src/runtime_api_impl/v5.rs

```diff
@@ -126,9 +132,9 @@ pub fn availability_cores<T: initializer::Config>() -> Vec<CoreState<T::Hash, Bl
 		.collect();
 
 	// This will overwrite only `Free` cores if the scheduler module is working as intended.
-	for scheduled in <scheduler::Pallet<T>>::scheduled_claimqueue() {
-		core_states[scheduled.core.0 as usize] = CoreState::Scheduled(primitives::ScheduledCore {
-			para_id: scheduled.paras_entry.para_id(),
+	for s in scheduled {
```

Comment:
This will break availability as it will override occupied cores as well and nodes won't do any availability work.

Something like this should help:
```
let overwrite = match core_states[s.core.0 as usize] {
			CoreState::Scheduled(_) | CoreState::Free => {
				true
			},
			CoreState::Occupied(occupied_core) => {
				if occupied_core.candidate_descriptor.para_id == s.paras_entry.para_id() {
					false
				} else {
					// A different para is being scheduled on this core
					true
				}
			}
		};

		if overwrite {
			core_states[s.core.0 as usize] = CoreState::Scheduled(primitives::ScheduledCore {
				para_id: s.paras_entry.para_id(),
				collator: None,
			});
		}
```

Reply:
I agree with the issue but the fix is probably wrong as well.

We should just not overwrite any already-occupied cores (their `next_up` should already be set appropriately). Overwriting only free cores seems fine and would address the issue we see (nothing is scheduled on the first block of the session).